cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -403,22 +403,22 @@ public synchronized void updateProperties(NamespaceId namespaceId, NamespaceMeta
   public NamespaceMeta get(NamespaceId namespaceId) throws Exception {
     Principal principal = authenticationContext.getPrincipal();
 
-    boolean isAuthorized = true;
+    UnauthorizedException lastUnauthorizedException = null;
     // if the principal is not same as cdap master principal do the authorization check. Otherwise, skip the auth check
     // See: CDAP-7387
     if (masterShortUserName == null || !masterShortUserName.equals(principal.getName())) {
       try {
         AuthorizationUtil.ensureAccess(namespaceId, authorizationEnforcer, principal);
       } catch (UnauthorizedException e) {
-        isAuthorized = false;
+        lastUnauthorizedException = e;
```

Comment:
Do we know why continue despite of unauthorized exception?

Mind adding a comment to give some explaination. Seems unintuitive.

Reply:
Basically, there seems to be an old feature where namespaces have owner principals which are recorded when they are created. There is a built-in exception check which automatically grants permissions if the person requesting access is the owner principal... we should consider deprecating this as somebody could create the namespace yet have their access to the namespace revoked.

I think it should be okay not to added a block comment because the code is already commented several lines below.